modules/management/zammad.nix

```diff
@@ -0,0 +1,59 @@
+{ config, pkgs, ... }:
+let
+  database_name = "zammad";
+  username = "zammad";
+  http_port = 3000;
+  ws_port = 3001;
+in
+{
+  sops.secrets.zammad_db_pass.owner = username;
+
+  services = {
+    postgresql = {
+      enable = true;
+      ensureDatabases = [ database_name ];
+      ensureUsers = [
+        {
+          name = username;
+          ensurePermissions = {
+            "DATABASE ${database_name}" = "ALL PRIVILEGES";
+          };
+        }
+      ];
+    };
+
+    zammad = {
+      enable = true;
+      package = pkgs.zammad;
+      port = http_port;
+      websocketPort = ws_port;
+      database = {
+        createLocally = false;
+        type = "PostgreSQL";
+        host = "/run/postgresql";
+        name = database_name;
+        user = username;
+        passwordFile = config.sops.secrets.zammad_db_pass.path;
```

Comment:
I dont see the point in the config where the database users password is set. 

(taken from vaultwarden)
```nix
systemd.services.vaultwarden-pgsetup = {
    description = "Prepare postgres database";
    wantedBy = [ "multi-user.target" ];
    after = [ "networking.target" "postgresql.service" ];
    serviceConfig.Type = "oneshot";

    path = [ pkgs.sudo config.services.postgresql.package ];
    script = ''
      # create postgres user with the specified password
      sudo -u ${config.services.postgresql.superUser} psql -c "ALTER ROLE vaultwarden WITH PASSWORD '$(cat ${config.sops.secrets.postgres_vaultwarden.path})'"
    '';
  };
```

alternatively, you also can put it into 

`systemd.services.postgresql.postStart`